### ISSUE TITLE
Extend protonuke capabilities

### ISSUE THREAD

**Issue by floren** [sandia-minimega/minimega#368]
There are some useful additions we could make to protonuke:

- Allow a single client machine to "spoof" an entire subnet by hand-crafting packets from different IPs.
- Add more protocols. We have http, https, smtp, and ssh. We could add telnet, ftp, DNS, NNTP (are your corporate users reading alt.ensign.wesley.die.die.die?), NTP, IMAP, IRC, 9P, and others.
- Let user specify a desired throughput that protonuke will attempt to achieve over an N-second sliding window.

Any more suggestions? I'll happily jump on any of these, but prioritization would be helpful. I'd say throughput would be a good start.

**Comment by djfritz** [comment]
- improved logging, including syslog output. Logging should be geared towards building network statistics in things like splunk
- custom certificates for tls enabled services (this is already in another ticked #413)

**Comment by jcrussell** [comment]
It would be cool to have `miniweb` that generates a more realistic web. VMs would each host many sites that are all interconnected via links with hostnames resolved via DNS. I'm envisioning `miniweb` running inside the VMs and communicating with other instances via `meshage`. We could then have protonuke implement a random surfer.

**Comment by djfritz** [comment]
We talked about this once before and I can't remember why we nixed it. 

- protonuke has a random surfer already
- the protonuke-protonuke coordination I like, but there's an alternative of coordinating in minimega and simply starting protonuke slaves. This should be a topic on it's own probably.
- the virtual hosts idea is certainly something we should support eventually (and we have implemented in go already somewhere else), since real web servers do this too. For authentic content, we have the `-httproot` flag so the user can provide a directory of content.
